src/WriteClient.ts

```diff
@@ -0,0 +1,1323 @@
+								}
+
+								// `masterLanguageDocument` is a new document
+								if (!masterLanguageDocumentID) {
+									masterLanguageDocumentID = documents.get(
+										masterLanguageDocument,
+									)?.id
+								}
+							}
+						} else {
+							masterLanguageDocumentID = params.masterLanguageDocument.id
+						}
+					} else if (document.alternate_languages) {
+						masterLanguageDocumentID = document.alternate_languages.find(
+							({ lang }) => lang === masterLocale,
+						)?.id
+					}
+				}
+
+				const { id } = await this.createDocument(
+					// We'll upload docuements data later on.
+					{ ...document, data: {} },
+					params.documentName,
+					{
+						masterLanguageDocumentID,
+						...fetchParams,
+					},
+				)
+
+				// Index old ID for Prismic to Prismic migration
+				if (document.id) {
+					documents.set(document.id, { ...document, id })
+				}
+				documents.set(document, { ...document, id })
+			}
+		}
+
+		reporter?.({
+			type: "documents:created",
+			data: {
+				created,
+				documents,
+			},
+		})
+
+		return documents
+	}
+
+	/**
+	 * Updates documents in the Prismic repository's migration release with their
+	 * patched data.
+	 *
+	 * @param migration - A migration prepared with {@link createMigration}.
+	 * @param assets - A map of assets available in the Prismic repository.
+	 * @param documents - A map of documents available in the Prismic repository.
+	 * @param params - An event listener and additional fetch parameters.
+	 *
+	 * @internal This method is one of the step performed by the {@link migrate} method.
+	 */
+	private async migrateUpdateDocuments(
+		migration: Migration<TDocuments>,
+		assets: AssetMap,
+		documents: DocumentMap<TDocuments>,
+		{
+			reporter,
+			...fetchParams
+		}: { reporter?: (event: MigrateReporterEvents) => void } & FetchParams = {},
+	): Promise<void> {
+		let i = 0
+		for (const { document, params } of migration.documents) {
+			reporter?.({
+				type: "documents:updating",
+				data: {
+					current: ++i,
+					remaining: migration.documents.length - i,
+					total: migration.documents.length,
+					document,
+					documentParams: params,
+				},
+			})
+
+			const { id, uid } = documents.get(document)!
+			const data = await patchMigrationDocumentData(
+				document.data,
+				assets,
+				documents,
+			)
+
+			await this.updateDocument(
+				id,
+				// We need to forward again document name and tags to update them
+				// in case the document already existed during the previous step.
+				{ documentName: params.documentName, uid, tags: document.tags, data },
+				fetchParams,
+			)
+		}
+
+		reporter?.({
+			type: "documents:updated",
+			data: {
+				updated: migration.documents.length,
+			},
+		})
+	}
+
+	/**
+	 * Queries assets from the Prismic repository's media library.
+	 *
+	 * @param params - Parameters to filter, sort, and paginate results.
+	 *
+	 * @returns A paginated response containing the result of the query.
+	 */
+	async getAssets({
```

Comment:
It could be made private I think(?), I don't really see a real valid use case for it apart from perhaps migrating an image-heavy website where you already have put the assets in the media library, tagged them, and then want to assign them programmatically to your documents 🤔

But yeah, feels far-fetched~

Reply:
Mark as private and comment it down